scrnaseq.Rmd

```diff
@@ -1761,8 +1768,56 @@ if (markers_found) {
 }
 ```
 
-### Heatmaps of differentially expressed genes {.tabset}
-#### All up- and down-regulated genes
+## Dotplots per cluster per sample 
+If the dataset contains multiple samples, we can visualise the expression of a gene that is up-regulated in a cluster separately for each sample. For each cluster, we extract up-regulated genes, and visualise expression of these genes in all cells in that cluster, split by their sample of origin. 
+```{r degs_plot_dotplotpercl_height}
+fig_height_degs_per_cl = max(5, 
+                             max(2, 0.3 * (sc$orig.ident %>% unique() %>% length())) * length(levels(sc$seurat_clusters)) * 2)
+```
+
+First, we plot scaled expression as explained above (see section Known marker genes). This plot allows us to judge whether the expression of a gene is increased in one sample as compared to the other samples.
+```{r degs_plot_dotplotpercl_scaled, fig.height=fig_height_degs_per_cl}
+n_genes_max_dotplot = 50
```

Comment:
That could be a parameter... . Or hard-code it

Reply:
It is hard-coded in this variable as I use it twice in the code section. I don't this should be a parameter though for the parameter section.

Reply:
Changed it, but now 50 comes up twice as hard-coded.

Reply:
No, never mind, I found another solution.